Make M117 message persistent on LCD when used with M601 or G4

gudnimg opened the pull request:
If M117 is called before M601 or G4 then we would like the custom message to
be visible on the LCD screen until the print is resumed.

Change in memory: +16 bytes of flash

Fixes #3316 (M601)
Fixes #1443 (G4)
Fixes #2970 (G4)

Video demonstration (with M601):

* Before (current Master branch): https://www.youtube.com/watch?v=8Mr-1lYwjsk
* After (this PR): https://www.youtube.com/watch?v=A7_uLBHg0qE

gudnimg committed:
Make M117 custom message persistant if print is paused

If M117 is called before M601 then we would like the custom message to
be visible on the LCD screen until the print is resumed.

Change in memory: +10 bytes of flash

Fixes #3316

leptun commented on Firmware/ultralcd.cpp:
```diff
@@ -676,6 +676,9 @@ void lcdui_print_status_line(void)
         case CustomMsg::Resuming: //Resuming
             lcd_puts_at_P(0, 3, _T(MSG_RESUMING_PRINT));
             break;
+        case CustomMsg::M117:
```
Merge this case with these:
```
        case CustomMsg::MsgUpdate: //Short message even while printing from SD
        case CustomMsg::Status: // Nothing special, print status message normally
        case CustomMsg::M0Wait: // M0/M1 Wait command working even from SD
```
They are all identical and could probably be optimized better this way.

gudnimg replied:
Done

leptun commented on Firmware/Marlin_main.cpp:
```diff
@@ -4272,7 +4272,7 @@ void process_commands()
             *(starpos) = '\0';
         lcd_setstatus(strchr_pointer + 5);
-        custom_message_type = CustomMsg::MsgUpdate;
+        custom_message_type = CustomMsg::M117;
```
CustomMsg::MsgUpdate and CustomMsg::M117 can be one and the same.

gudnimg replied:
Nice, didn't notice that!

gudnimg committed:
Replace CustomMsg::MsgUpdate with CustomMsg::M117

MsgUpdate was only used with M117, renaming the message type to M117 may be more descriptive for current use case.

gudnimg commented:
@leptun I renamed `CustomMsg::MsgUpdate` to `CustomMsg::M117` and reduced flash by 2 bytes.

gudnimg committed:
Make M117 message persistent when using G4 command

This is also supported in Marlin 2 FW

Fixes #1443
Fixes #2970

gudnimg commented:
I added a commit which fixes the same issue when using `M117` with  `G4` command since it is supported by Marlin 2. Only adds 8 bytes more.

leptun reviewed (approved):
Seems to work nicely, except for one issue: The lcd_status_message gets cleared on resume. Probably not a serious issue, but I had to mention it.
